scripts/us_census/pep/Population_Estimate_by_ASR/national_1980_1989.py

```diff
@@ -0,0 +1,128 @@
+    '''
+    df['Year'] = "19" + df['1'].str[1:3]
+    df['Age'] = df['2'].astype(float).astype(int)
+    df['Male'] = df['4']
+    df['Female'] = df['5']
+    df['WhiteAggrAll'] = df['6'] + df['7']
+    df['Male_WhiteAll'] = df['6']
+    df['Female_WhiteAll'] = df['7']
+    df['BlackOrAfricanAmericanAggrAll'] = df['8'] + df['9']
+    df['Male_BlackOrAfricanAmericanAll'] = df['8']
+    df['Female_BlackOrAfricanAmericanAll'] = df['9']
+    df['AmericanIndianAndAlaskaNativeAggrAll'] = df['10'] + df['11']
+    df['Male_AmericanIndianAndAlaskaNativeAll'] = df['10']
+    df['Female_AmericanIndianAndAlaskaNativeAll'] = df['11']
+    df['AsianOrPacificIslanderAggr'] = df['12'] + df['13']
+    df['Male_AsianOrPacificIslander'] = df['12']
+    df['Female_AsianOrPacificIslander'] = df['13']
+    df = df.drop([
+        "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12",
+        "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23"
+    ],
+                 axis=1)
+    return df
+
+
+def national1980():
+    '''
+    This Python Script Loads csv datasets
+    from 1980-1989 on a National Level,
+    cleans it and create a cleaned csv.
+    '''
+    # Getting list of URLs from JSON file.
+    _ZIP_DIR = os.path.dirname(
+        os.path.abspath(__file__)) + os.sep + 'Zip1980-90'
+    _urls = _input_url("national.json", "1980-90")
+
+    # Creation of a folder if it does not exist.
+    if not os.path.exists(_ZIP_DIR):
+        os.mkdir(_ZIP_DIR)
+    current_dir = os.path.dirname(os.path.abspath(__file__)) + os.sep
+    os.chdir(_ZIP_DIR)
+
+    # Extraction of ZIP files.
+    for url in _urls:
+        with urlopen(url) as resp:
+            # unzipping the dataset
+            with ZipFile(BytesIO(resp.read()), 'r') as zipfile:
+                zipfile.extractall()
+    _urls = _urls = _input_url("national.json", "1980-90files")
+
+    cols = ["0", "1", "2", "3", "4", "5", "6", "7",\
+                        "8", "9", "10", "11","12", "13", "14", "15",\
+                        "16", "17", "18", "19", "20", "21", "22"]
+    # Used to collect data after every loop for every file's df.
+    final_df = pd.DataFrame()
+
+    for file in _urls:
+
+        # Reading the txt format input file converting it to a dataframe.
+        # Delimitng the columns by whitespace.
+        df = pd.read_table(file,
+                           index_col=False,
+                           delim_whitespace=True,
+                           engine='python',
+                           names=cols)
+        df = df.loc[df['0'].isin(['2I', '9P'])]
+        df['1'] = df['1'].astype(str)
+        df1 = df[df['1'].str.contains("100")]
+        df = df[~df['1'].str.contains("999")]
+        df = df[~df['1'].str.contains("100")]
+        for i in range(22, 1, -1):
+            j = i + 1
+            i = str(i)
+            j = str(j)
+            df1[j] = df1[i]
+            i = int(i)
+        df1['2'] = df1['1'].str[-5:]
```

Comment:
Some of these seem like magic numbers (why -5?). Either use helpful comments or define these as contants.

Reply:
Here, and everywhere else too, please.

Reply:
Added helpful comments where needed.